workflow/workflow.go

```diff
@@ -470,6 +470,34 @@ func UpsertSearchAttributes(ctx Context, attributes map[string]interface{}) erro
 	return internal.UpsertSearchAttributes(ctx, attributes)
 }
 
+// UpsertMemo is used to add or update workflow memo.
+// UpsertMemo will merge keys to the existing map in workflow. For example:
+//
+//	func MyWorkflow(ctx workflow.Context, input string) error {
+//		memo1 := map[string]interface{}{
+//			"Key1": 1,
+//			"Key2": true,
+//		}
+//		workflow.UpsertMemo(ctx, memo1)
+//
+//		memo2 := map[string]interface{}{
+//			"Key1": 2,
+//			"Key3": "seattle",
+//		}
+//		workflow.UpsertMemo(ctx, memo2)
+//	}
+//
+// The workflow memo will eventually be:
+//
+//	map[string]interface{}{
+//		"Key1": 2,
+//		"Key2": true,
+//		"Key3": "seattle",
+//	}
+func UpsertMemo(ctx Context, memo map[string]interface{}) error {
```

Comment:
We should probably make it clear not all servers support this (and maybe say which server version does)